### PR TITLE
[LUCENE-6687] not necessary nested for loop removed for terms retriev…

### PR DESCRIPTION
**Pull request by alessandrobenedetti**
Bug in term frequencies calculation for the MLT

### CODE REVIEW

File: lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java [hunk 1]
```diff
@@ -186,6 +200,117 @@ public void testMultiValues() throws Exception {
...
   }
 
+  public void testLiveMapDocument_minTermFrequencySet_shouldBuildQueryAccordingToCorrectTermFrequencies() throws Exception {
+    MoreLikeThis mlt = new MoreLikeThis(reader);
+    Analyzer analyzer = new MockAnalyzer(random(), MockTokenizer.WHITESPACE, false);
+    mlt.setAnalyzer(analyzer);
+    mlt.setMinDocFreq(0);
+    mlt.setMinTermFreq(3);
+    mlt.setMinWordLen(1);
+    String sampleField1 = "text";
```
Comment (alessandrobenedetti) [inline lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java]: move to class constant

File: lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java [hunk 2]
```diff
@@ -186,6 +200,117 @@ public void testMultiValues() throws Exception {
...
+    String textValue1 = "apache apache lucene lucene";
+    String textValue2 = "apache2 apache2 lucene2 lucene2 lucene2";
+    filteredDocument.put(sampleField1, Arrays.asList(textValue1));
+    filteredDocument.put(sampleField2, Arrays.asList(textValue2));
+
+    BooleanQuery query = (BooleanQuery) mlt.like(filteredDocument);
+    Collection<BooleanClause> clauses = query.clauses();
+
+    HashSet<Term> unexpectedTerms = new HashSet<>();
+    unexpectedTerms.add(new Term("text", "apache"));//Term Frequency < Minimum Accepted Term Frequency
```
Comment (alessandrobenedetti) [inline lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java]: reference sampleField1 constant

File: lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java [hunk 3]
```diff
@@ -186,6 +200,117 @@ public void testMultiValues() throws Exception {
...
+    assertEquals("Expected 0 clauses only!", 0, clauses.size());
+
+    analyzer.close();
+  }
+
+  public void testLiveMapDocument_queryFieldsSet_shouldBuildQueryFromSpecifiedFieldnamesOnly() throws Exception {
+    MoreLikeThis mlt = new MoreLikeThis(reader);
+    Analyzer analyzer = new MockAnalyzer(random(), MockTokenizer.WHITESPACE, false);
+    mlt.setAnalyzer(analyzer);
+    mlt.setMinDocFreq(1);
```
Comment (alessandrobenedetti) [inline lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java]: put it to 0

File: lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java [hunk 4]
```diff
@@ -186,6 +200,117 @@ public void testMultiValues() throws Exception {
...
+    BooleanQuery query = (BooleanQuery) mlt.like(filteredDocument);
+    Collection<BooleanClause> clauses = query.clauses();
+    HashSet<Term> clausesTerms = new HashSet<>();
+    for (BooleanClause clause : clauses) {
+      Term term = ((TermQuery) clause.getQuery()).getTerm();
+      clausesTerms.add(term);
+    }
+    assertEquals("Expected 2 clauses only!", 2, clauses.size());
+
+    HashSet<Term> expectedTerms = new HashSet<>();
```
Comment (alessandrobenedetti) [inline lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java]: move to the beginning

File: lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java [hunk 5]
```diff
@@ -186,6 +200,117 @@ public void testMultiValues() throws Exception {
...
+    expectedTerms.add(new Term("text", "lucene"));
+
+    HashSet<Term> unexpectedTerms = new HashSet<>();
+    unexpectedTerms.add(new Term("text", "apache2"));
+    unexpectedTerms.add(new Term("text", "lucene2"));
+
+    //None of the Not Expected terms is in the query
+    for (BooleanClause clause : clauses) {
+      Term term = ((TermQuery) clause.getQuery()).getTerm();
+      assertFalse("Unexpected term '" + term + "' found in query terms", unexpectedTerms.contains(term));
```
Comment (alessandrobenedetti) [inline lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java]: move clauses terms accumulation here

File: lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java [hunk 6]
```diff
@@ -186,6 +200,117 @@ public void testMultiValues() throws Exception {
...
+    }
+
+    //All of the Expected terms are in the query
+    for (Term expectedTerm : expectedTerms) {
+      assertTrue("Expected term '" + expectedTerm + "' is not found in query terms", clausesTerms.contains(expectedTerm));
+    }
+
```
Comment (alessandrobenedetti) [inline lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java]: Add this -> All of the terms in the query must be expected

**Comment by janhoy** [comment]
Closing since JIRA is closed. Please remember to close PRs when closing JIRAs

**Pull request closed without merging**